j2v/generation/generator.py

```diff
@@ -152,6 +153,8 @@ def __add_dimension(self, field_path_sql, current_view, dimension_name, dim_val,
 
         dimension_name_final = "_".join(nice_dimension_name)
 
+        primary_key_field = "\n    primary_key: yes" if self.primary_key is not None and dimension_name_final in self.primary_key else ""
```

Comment:
1. if dimension_name_final is `id` and primary_key is `order_id` things are broken
2. A user specifies primary_key as a name of key inside JSON, so we should look into key/path, not the dimension_name which is created by the code.

Reply:
Thanks!